src/wp-includes/link-template.php

```diff
@@ -4230,7 +4230,7 @@ function get_avatar_data( $id_or_email, $args = null ) {
 			'default'        => get_option( 'avatar_default', 'mystery' ),
 			'force_default'  => false,
 			'rating'         => get_option( 'avatar_rating' ),
-			'scheme'         => null,
+			'scheme'         => 'https',
```

Comment:
This may cause problems for sites using hooks such as `pre_get_avatar_data` to replace gravatars with their own system. If the replacement system doesn't support https then the images will display as broken.

How about forcing gravatar.com urls to HTTPs regardless of this setting and adding in a comment explaining that gravatar.com redirects all URLs to SSL so the setting is disregarded for that reason.